indexer/feature_data.cpp

```diff
@@ -262,20 +263,37 @@ bool FeatureParams::AddHouseName(string const & s)
   return false;
 }
 
-bool FeatureParams::AddHouseNumber(string const & ss)
+bool FeatureParams::AddHouseNumber(string houseNumber)
 {
-  if (!feature::IsHouseNumber(ss))
+  strings::Trim(houseNumber);
+  if (houseNumber.empty())
     return false;
 
-  // Remove trailing zero's from house numbers.
+  // Negative house numbers are not supported.
+  if (houseNumber.front() == '-')
+    return false;
+
+  // Replace full-width digits, mostly in Japan, by ascii-ones.
+  strings::NormalizeDigits(houseNumber);
+
+  // Remove trailing zeroes from house numbers.
   // It's important for debug checks of serialized-deserialized feature.
-  string s(ss);
-  uint64_t n;
-  if (strings::to_uint64(s, n))
-    s = strings::to_string(n);
+  for (size_t i = 0; i < houseNumber.size(); ++i)
+  {
+    if (houseNumber[i] != '0')
```

Comment:
In any case you can both combine leading zeroes removal with check that the string has at least one digit. Moreover, the code proposed by me can be easily modified to do this.